### PR TITLE
refactor the logic of data proxy refresher

### LINKED ISSUE #586: refactor the logic of data proxy refresher
Currently, there is a lot of code in the main function. Use function decomposition to make it more readable.

### PR DESCRIPTION
**Pull request by maxsarratt**
resolves #586

### CODE REVIEW

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go [hunk 1]
```diff
@@ -25,231 +26,421 @@ func init() {
...
+}
+
+func (r *DataproxyRefresher) getMapping(
+	schemaName string,
+) (map[string]string, error) {
 	mapping, err := importutil.GetMapping(schemaName)
 	if err != nil {
-		logger.Error("Failed to get mapping.", err)
-		cleanUp()
+		return nil, fmt.Errorf("failed to get mapping: %w", err)
```
Comment (maxsarratt) [inline services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:117]: @TingSyuanWang, the new code will exit with a status of 1 if there's any error, including failure to find the mapping. I'm not sure if we should change the behavior, such as ignoring the not found errors.

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go [hunk 2]
```diff
@@ -25,231 +26,421 @@ func init() {
...
+		)
+	}
 
-			if profileHash != profile.SourceDataHash {
-				logger.Info(
-					"Source data hash mismatch: " + profile.Cuid + " - " + profile.Oid + " : " + profile.SourceDataHash + " - " + profileHash,
-				)
-
-				// reconstruct data
-				profileJSON, err = importutil.MapProfile(
```
Comment (TingSyuanWang) [inline services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:122]: I believe these deleted lines are crucial component.
The profileJSON we generated in the past only mapped the fields and values.
Right before updating the profile in the Index, it's imperative that we incorporate `importutil.MapProfile` to include additional information for converting KVM to organizational schemas.

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go [hunk 3]
```diff
@@ -25,231 +26,421 @@ func init() {
...
-				}
-
-				// save node_id to profile
-				err = profileSvc.UpdateNodeID(oid, nodeID)
-				if err != nil {
-					logger.Error("Update node id failed. Profile ID: "+oid, err)
-					cleanUp()
-				}
-			} else {
-				err = svc.UpdateAccessTime(profile.Oid)
```
Comment (TingSyuanWang) [inline services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:210]: @geoffturk @maxsarratt 

I've observed that the logic in this section has been removed, and I'd like to verify if this deletion is intentional and correct.
The logic here is that when there are no updates to the profile, we should update the access_time. However, I believe we've removed this practice.
If you can confirm that this is indeed the case, please indicate your agreement with a thumbs-up 👍.

Comment (geoffturk) [reply services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:210]: We should not delete this access time update. It is done to show in the metadata when was the last time we checked the profile data source for any changes to the profile. Even if nothing changed, we still want to update this timestamp. It is different from the `last_updated` timestamp that we store in every profile in the index.

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go (hunk 2 above)
Comment (geoffturk) [reply services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:122]: That's right. We need to keep the profile mapping here.

**Review by geoffturk — CHANGES_REQUESTED** [review]
See the comments ☝️

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go (hunk 3 above)
Comment (maxsarratt) [reply services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:210]: The logic resides in the `updateProfileAccessTime` method. When hashedData is not equal to profile.SourceDataHash, we will call this method.

File: services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go (hunk 2 above)
Comment (maxsarratt) [reply services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go:122]: @TingSyuanWang I have updated it. Please have check.

**Review by TingSyuanWang — APPROVED** [review]
LGTM 🚀

**Review by geoffturk — APPROVED** [review]
LGTM 🚀

**Pull request merged**